hugegraph-example/src/main/java/com/baidu/hugegraph/example/CollectionIndexExample.java

```diff
@@ -0,0 +1,97 @@
+import com.baidu.hugegraph.HugeGraph;
+import com.baidu.hugegraph.schema.SchemaManager;
+import com.baidu.hugegraph.testutil.Assert;
+import com.baidu.hugegraph.traversal.optimize.ConditionP;
+import com.baidu.hugegraph.util.Log;
+import com.google.common.collect.ImmutableSet;
+
+public class CollectionIndexExample {
+
+    private static final Logger LOG = Log.logger(CollectionIndexExample.class);
+
+    public static void main(String[] args) throws Exception {
+        LOG.info("TestIndexExample start!");
+
+        HugeGraph graph = ExampleUtil.loadGraph();
+
+        initData(graph);
+        queryTest(graph);
+
+        graph.close();
+
+        HugeFactory.shutdown(30L);
+    }
+
+    public static void initData(final HugeGraph graph) {
+        SchemaManager schema = graph.schema();
+
+        schema.propertyKey("name").asText().create();
+        schema.propertyKey("tags").asText().valueSet().create();
+        schema.propertyKey("category").asText().valueSet().create();
+        schema.propertyKey("country").asText().create();
+
+        schema.vertexLabel("soft").properties("name", "tags", "country", "category")
+              .primaryKeys("name").create();
+
+        schema.indexLabel("softByTag").onV("soft").secondary()
+              .by("tags").create();
+        schema.indexLabel("softByCategory").onV("soft").search()
+              .by("category").create();
+
+        graph.addVertex(T.label, "soft", "name", "hugegraph",
+                        "country", "china",
+                        "category", Arrays.asList("graphdb", "db"),
+                        "tags", Arrays.asList("graphdb", "gremlin"));
+
+        graph.addVertex(T.label, "soft", "name", "neo4j",
+                        "country", "usa",
+                        "category", Arrays.asList("graphdb", "db"),
+                        "tags", Arrays.asList("graphdb", "cypher"));
+
+        graph.addVertex(T.label, "soft", "name", "janusgraph",
+                        "country", "usa",
+                        "category", Arrays.asList("graphdb", "db"),
+                        "tags", Arrays.asList("graphdb", "gremlin"));
+
+        graph.tx().commit();
+    }
+
+    public static void queryTest(final HugeGraph graph) {
+        List<Vertex> vertices;
+        vertices = graph.traversal().V().has("soft", "category",
+                                             ConditionP.textContains("graphdb")).toList();
+        Assert.assertEquals(3, vertices.size());
+
+        // query by single item
+        vertices = graph.traversal().V().has("soft", "tags", "gremlin").toList();
+        Assert.assertEquals(2, vertices.size());
+
+        vertices = graph.traversal().V().has("soft", "tags", "graphdb").toList();
+        Assert.assertEquals(3, vertices.size());
+
+        // query by contains
+        vertices = graph.traversal().V().has("soft", "tags", ConditionP.contains("gremlin")).toList();
+        Assert.assertEquals(2, vertices.size());
+
+        // add a new tag
```

Comment:
may be we can separate test by 3 case:
1. write property index & read property by index  (read / write)
2. update property for `set` like `[a, b]` --> `[a, b, c]`, then assert   (update)
   - if we add property `c`, it should be read by index
   - if we remove property `a`, it shouldn't be read by index
3. delete vertex / edge, then assert read from index `size == 0` (delete)